### PR TITLE
feat(api): Migrate DB Labware into V2 Format

### LINKED ISSUE #3542: Apiv2 Backcompat: Migrate custom labware from sqlite to labware v2 format
Though standard labware can be looked up using the equivalency table mentioned in #3541 , custom labware created and saved in the sqlite database cannot. Instead, scan the database for all non-standard labware and dump it to labware v2 format in the normal custom labware directory. Afterwards, delete the database - its presence serves as a marker of whether the migration has been performed.

### PR DESCRIPTION
**Pull request by Laura-Danielle**
## overview
Closes #3542. Any custom labware from the API V1 database should now be migrated into API V2 format under the custom labware location with the `legacy_api` as the namespace.

## changelog

- add a migration function which filters out labwares and only creates unfound labwares in v2 with a new namespace called `legacy_api`
- Migration function can get called either on api server start or the simulate/execute entrypoints

## review requests

@IanLondon if you have time, sanity checking these definitions would be helpful. For others testing, please set `MIGRATE_V1_LABWARE` on the robot's OS environment variable (etc/profile.d/ot-environ.sh) and then load the labware using the `namespace` key in api v2. Once the labware load backcompat function is fully implemented we can retest loading the labware.

### CODE REVIEW

File: api/src/opentrons/simulate.py [hunk 1]
```diff
@@ -392,6 +395,18 @@ def main() -> int:
     parser = get_arguments(parser)
 
     args = parser.parse_args()
+    if os.environ.get('MIGRATE_V1_LABWARE') and\
```
Comment (sfoster1) [inline api/src/opentrons/simulate.py]: as with `execute`, centralize this logic elsewhere

File: api/tests/opentrons/protocol_api/test_legacy_migration.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+@pytest.fixture
+def labware_fixture():
+    return {
+        'trough-12row-short': db_cmds.load_container('trough-12row-short'),
+        '48-vial-plate': db_cmds.load_container('48-vial-plate'),
+        '24-vial-rack': db_cmds.load_container('24-vial-rack')}
+
+
+@pytest.mark.api2_only
+def test_migrated_labware_shape(monkeypatch, config_tempdir, labware_fixture):
```
Comment (sfoster1) [inline api/tests/opentrons/protocol_api/test_legacy_migration.py]: we should also do a schema check here with `jsonschema`

Comment (sfoster1) [reply api/tests/opentrons/protocol_api/test_legacy_migration.py]: and run this on probably _all_ the labware we know will be migrated as well as a couple special irregular custom labwares from `labware.create()` (maybe investigate examples from the protocols repo?)

File: api/src/opentrons/protocol_api/legacy_wrapper/api.py [hunk 3]
```diff
@@ -244,3 +205,22 @@ def build_globals(context: 'ProtocolContext'):
     rob._set_globals(instr, lw, mod)
 
     return {'robot': rob, 'instruments': instr, 'labware': lw, 'modules': mod}
+
+
+def maybe_migrate_containers():
+    result = False
+    if os.environ.get('MIGRATE_V1_LABWARE') and\
```
[the remark this answers is not in the document]
Comment (Laura-Danielle) [reply api/src/opentrons/protocol_api/legacy_wrapper/api.py]: yep

File: api/src/opentrons/protocol_api/legacy_wrapper/containers_wrapper.py [hunk 4]
```diff
@@ -1,10 +1,200 @@
...
+    if well_props.get('diameter'):
+        lw_dict['wells'][well_name]['diameter'] = well_props.get('diameter')
+        lw_dict['wells'][well_name]['shape'] = 'circular'
+    else:
+        lw_dict['wells'][well_name]['xDimension'] = well_props.get('length')
+        lw_dict['wells'][well_name]['yDimension'] = well_props.get('width')
+        lw_dict['wells'][well_name]['shape'] = 'rectangular'
+
+
+def create_new_labware_definition(labware: Container, labware_name: str):
```
[the remark this answers is not in the document]
Comment (sfoster1) [reply api/src/opentrons/protocol_api/legacy_wrapper/containers_wrapper.py]: Read this a bit more closely and saw the `converted_labware_name`. Let's make that a helper function.

**Pull request merged**